components/menu/menu/Menu.tsx

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {
+    useRef,
+    useState,
+    useEffect,
+    useCallback,
+} from 'react';
+
+import {CSSTransition} from 'react-transition-group';
+
+import {useClickOutsideRef} from 'components/global_header/hooks';
+
+import MenuItem from './MenuItem';
+import MenuData from './MenuData';
+import styled, { css } from 'styled-components';
+
+const Overlay = styled.div(() => {
+    return css`
+        background-color: rgba(0, 0, 0, 0);
+        display: flex;
+        align-items: flex-end;
+        justify-content: center;
+        height: 100%;
+        min-height: 100%;
+        left: 0;
+        right: 0;
+        top: 0;
+        position: fixed;
+        overflow: auto;
+        -webkit-overflow-scrolling: touch;
+        overscroll-behavior: contain;
+        pointer-events: auto;
+        -ms-scroll-chaining: none;
+        transition: 1s;
+        transition-property: background-color;
+        transition-timing-function: ease-in-out;
+
+        &.fade-enter {
+            background-color: rgba(0, 0, 0, 0);
+        }
+
+        &.fade-enter-active {
+            background-color: rgba(0, 0, 0, 0.5);
+        }
+
+        &.fade-enter-done {
+            background-color: rgba(0, 0, 0, 0.5);
+        }
+
+        &.fade-exit {
+            background-color: rgba(0, 0, 0, 0.5);
+        }
+
+        &.fade-exit-active {
+            background-color: rgba(0, 0, 0, 0);
+        }
+
+        &.fade-exit-done {
+            background-color: rgba(0, 0, 0, 0);
+        }
+    `;
+});
```

Comment:
You can use the "easy" syntax here, since you are not using any props inside:

```suggestion
const Overlay = styled.div`
    background-color: rgba(0, 0, 0, 0);
    display: flex;
    align-items: flex-end;
    justify-content: center;
    height: 100%;
    min-height: 100%;
    left: 0;
    right: 0;
    top: 0;
    position: fixed;
    overflow: auto;
    -webkit-overflow-scrolling: touch;
    overscroll-behavior: contain;
    pointer-events: auto;
    -ms-scroll-chaining: none;
    transition: 1s;
    transition-property: background-color;
    transition-timing-function: ease-in-out;

    &.fade-enter {
        background-color: rgba(0, 0, 0, 0);
    }

    &.fade-enter-active {
        background-color: rgba(0, 0, 0, 0.5);
    }

    &.fade-enter-done {
        background-color: rgba(0, 0, 0, 0.5);
    }

    &.fade-exit {
        background-color: rgba(0, 0, 0, 0.5);
    }

    &.fade-exit-active {
        background-color: rgba(0, 0, 0, 0);
    }

    &.fade-exit-done {
        background-color: rgba(0, 0, 0, 0);
    }
`;
```